Add New Teleporting Boss that Shoots Projectiles

Lukas-bit opened the pull request:
Added a new boss (that is not yet placed on the map) that will frequently fire projectiles at the player, and will often teleport away from the player when they are within a certain distance from the boss.

Lukas-bit committed:
Merge branch 'master' of https://github.com/noahwagner04/CSE320_Game

Lukas-bit committed:
Merge branch 'master' into lukas/fugacious_wraith

Lukas-bit committed:
Add New Boss That Shoots Projectiles

Added a new boss that frequently shoots projectiles. The current idea is
that the boss will be stationary most of the time, and then will
teleport away when the player gets close (on a cooldown, of course).
Teleportation feature not yet implemented

Lukas-bit committed:
Add Teleportation Ability to Wraith Boss

The boss now frequently teleports away from the player when the player
is within a certain distance and after a set cooldown

Lukas-bit committed:
Remove While Loop for Checking Player Distance

Removed while loop that checked for player distance, and also uses the
distance_to() method to check for relative distance instead of comparing
position vectors. Also, increased speed at which the boss shoots
projectiles and increased projectile velocity. Additionally, the boss
will teleport more frequently

obimax99 reviewed (changes requested):
Looks good, you should be able to make the projectile extend Projectile rather than Node2D and it then won't need to have all those lines of code

Lukas-bit committed:
Extend Projectile and Add Second Phase

Instead of extending Node2D, the wraith projectile now extends the
Projectile class. Also, I added a second phase to the boss, which as of
now just halves the cooldown period between projectile shots and
teleportations.
